### PR TITLE
THRIFT-5600: rust lib to upgrade to edition 2021

### PR DESCRIPTION
**Pull request by Jimexist**
upgrade rust toolchain to 1.50 to fix some compiling issues

- closes #2627
- closes https://github.com/apache/thrift/pull/2527

- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?  (not required for trivial changes)
- [x] If a ticket exists: Does your pull request title follow the pattern "THRIFT-NNNN: describe my issue"?
- [x] Did you squash your changes to a single commit?  (not required, but preferred)
- [x] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [x] If your change does not involve any code, include `[skip ci]` anywhere in the commit message to free up build resources.

### CODE REVIEW

**Comment by Jimexist** [comment]
before:
- #2527 
let's do this because it requires no gen code changes

File: lib/rs/src/server/multiplexed.rs [hunk 1]
```diff
@@ -170,7 +170,7 @@ fn split_ident_name(ident_name: &str) -> (Option<&str>, &str) {
             let (_, svc_call) = svc_call.split_at(1); // remove colon from service call name
             (Some(svc_name), svc_call)
         })
-        .or_else(|| Some((None, ident_name)))
+        .or(Some((None, ident_name)))
```
Comment (Jimexist) [inline lib/rs/src/server/multiplexed.rs:173]: clippy fix

File: lib/rs/src/server/multiplexed.rs [hunk 2]
```diff
@@ -272,10 +272,10 @@ mod tests {
             _: &mut dyn TInputProtocol,
             _: &mut dyn TOutputProtocol,
         ) -> crate::Result<()> {
-            let res = self
-                .invoked
-                .compare_and_swap(false, true, Ordering::Relaxed);
-            if res {
+            let res =
```
Comment (Jimexist) [inline lib/rs/src/server/multiplexed.rs:275]: API change

**Comment by Jimexist** [comment]
this somehow erred but i don't think they are triggered by the change

**Comment by Jens-G** [comment]
could you kick it to get another CI run please?

**Review by allengeorge — APPROVED** [review]
Thank you for this update, and I'm sorry it took me so long to get around to this.

**Comment by allengeorge** [comment]
I'm rerunning the build right now.

**Comment by allengeorge** [comment]
Suggest updating this to 1.61 as per conversation and also updating `THRIFT_ROOT/rust_toolchain`

**Comment by Jimexist** [comment]
i propose to merge this first - as the errors are Java related, which requires separate fixes. the master branch is now failing due to rust failures. this is really a seesaw situation.

**Comment by Jens-G** [comment]
@Jimexist could you possibly rebase and fix any conflicts?

**Comment by Jimexist** [comment]
@Jens-G and @allengeorge rebased now

**Pull request closed without merging**